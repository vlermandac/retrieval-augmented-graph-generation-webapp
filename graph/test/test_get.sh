#!/bin/bash

../KnowledgeGraph &
SERVER_PID=$!

sleep 2

JSON=$(cat <<EOF
{
  "index": "test_get",
  "triplet_lists": [
    {
      "id": 1,
      "triplets": [
        {"entity1": "A", "relation": "B", "entity2": "C"},
        {"entity1": "C", "relation": "D", "entity2": "E"}
      ]
    },
    {
      "id": 2,
      "triplets": [
        {"entity1": "A", "relation": "B", "entity2": "E"}
      ]
    }
  ]
}
EOF
)

# generating the graph at ../../data/test_get
curl -X POST -H "Content-Type: application/json" -d "$JSON" http://localhost:8080/generate
# getting the json graph as a response and saving it to tmp_get.json
curl -X POST -H "Content-Type: application/json" -d '{"index": "test_get", "values": []}' http://localhost:8080/get > tmp_get.json
# an empty 'values' list means that we want the whole graph

# removing stochastic values from the output json
sed '/"x":/d;/"y":/d;/"size"/d;' "tmp_get.json" > "output.json" 

expected=../../test/expected/test_get.json

sed '/"x":/d;/"y":/d;/"size"/d;' "$expected" > "expected.json"

if diff -B -Z "expected.json" "output.json"; then
  echo "TestGet passed"
  kill $SERVER_PID
  rm -rf ../../data/test_get
  exit 0
else
  echo "TestGet failed: files are different"
  kill $SERVER_PID
  rm -rf ../../data/test_get
  exit 1
fi
