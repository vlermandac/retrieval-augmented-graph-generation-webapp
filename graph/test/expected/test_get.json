{
  "attributes": {},
  "edges": [
    {
      "attributes": {
        "chunk_id": 1,
        "color": "#626880",
        "forceLabel": false,
        "label": "B",
        "size": 1
      },
      "source": "A",
      "target": "C"
    },
    {
      "attributes": {
        "chunk_id": 1,
        "color": "#626880",
        "forceLabel": false,
        "label": "D",
        "size": 1
      },
      "source": "C",
      "target": "E"
    },
    {
      "attributes": {
        "chunk_id": 2,
        "color": "#626880",
        "forceLabel": false,
        "label": "B",
        "size": 1
      },
      "source": "A",
      "target": "E"
    }
  ],
  "nodes": [
    {
      "attributes": {
        "color": "#babbf1",
        "id": 1,
        "label": "A",
        "size": 4,
        "x": 343.6574283768293,
        "y": 343.6503629388851
      },
      "key": "A"
    },
    {
      "attributes": {
        "color": "#babbf1",
        "id": 2,
        "label": "C",
        "size": 8,
        "x": 0.006687873664989389,
        "y": 486.0
      },
      "key": "C"
    },
    {
      "attributes": {
        "color": "#babbf1",
        "id": 3,
        "label": "E",
        "size": 12,
        "x": -343.6567895146397,
        "y": 343.65100185476115
      },
      "key": "E"
    }
  ],
  "options": {
    "multi": true
  }
}

